### ISSUE TITLE
Shiny is Stumping Me

### ISSUE THREAD

**Issue by LiamBurke24** [PecanProject/pecan#1667]
In trying to implement a file input tool on a shiny app, I am not seeing my input displayed as an output on the server side. 

## Description

I am testing this on my VM as test-pecan does not render shiny for some reason. 

## Context
Here is the relevant code: 
```
ui <- dashboardPage(
  
  dashboardHeader(title = "Data Ingest"),
  dashboardSidebar(
    sidebarMenu(
      menuItem("Import Data", tabName = "importData", icon = icon("file")),
      menuItem("Step 2 -- dbfiles record", tabName = "step2", icon = icon("cog")),
      menuItem("Step 3 -- format record", tabName = "step3", icon = icon("cog")),
      menuItem("Step 4 -- etc.", tabName = "step4", icon = icon("cog"))
    )
  ),
  dashboardBody(
    tabItems(
      tabItem(tabName = "importData",
              fluidRow( 
                box(
                  textInput("id", label = h3("Import From DataONE"), placeholder = "Enter doi or id here"),
                  actionButton(inputId = "D1Button", label = "Upload"),
                  hr(),
                  fluidRow(column(12, verbatimTextOutput("identifier"))) 
                ),
                
                box( #This is possibly a source of the error
                  # https://github.com/rstudio/shiny-examples/blob/master/009-upload/app.R
                  fileInput(inputId = "file", label = h3("Upload Local Files"), accept = NULL, multiple = TRUE, placeholder = "Drag and drop files here"),
                  tableOutput("contents")
                )
              )
      ),
      
      tabItem(tabName = "step2",
              h2("dbfiles tab content")
      )
      
      
    )
  )
)

server <- function(input, output) {
  options(shiny.maxRequestSize=30*1024^2) #maximum file input size
  
  # path <- PEcAn.utils::read_web_config("config.php")
    
   d1d <- eventReactive(input$D1Button, { PEcAn.data.land::dataone_download(input$id, filepath = "~/dump") }) #run dataone_download with input from id on click
  
  output$identifier <- renderText({
    d1d()
  })
  
  
###### FileInput
  output$contents <- renderTable({ #There must be something going wrong here
    # input$file1 will be NULL initially. After the user selects
    # and uploads a file, it will be a data frame with 'name',
    # 'size', 'type', and 'datapath' columns. The 'datapath'
    # column will contain the local filenames where the data can
    # be found.
    inFile <- input$file
    
    if (is.null(inFile))
      return(NULL)
    read.csv(inFile$datapath, header = TRUE)
    
  })

}

# Run the application 
shinyApp(ui = ui, server = server)

# example data: doi:10.6073/pasta/63ad7159306bc031520f09b2faefcf87
```
Any and all suggestions appreciated

**Comment by mdietze** [comment]
Ok, so I just pulled the code from PR #1648, specified the ```path``` variable (which isn't defined there) to be the ABSOLUTE path of something I know is writable, and gave the app a try. Dataone query went off without a hitch but then the wget failed because I'm on my Mac and don't have wget installed (oops, forgot I hadn't done that when upgrading machines).

So, I think the code probably works. My gut tells me that you should try specifying your path in absolute terms, not relative to home, since the web app may be running as the apache2 user not as carya

**Comment by LiamBurke24** [comment]
I am surprised that the quick test that Alexey put together to test for wget didn't catch it on your computer... And I will specify in absolute terms ( I forgot to change that here and use the path variable for the filepath...) Do you think that doing this will address what you said?

**Comment by ashiklom** [comment]
Closing this because I think it was addressed between #1648, #1664, and #1666.
